opal_common/git/repo_cloner.py

```diff
@@ -59,34 +59,69 @@ def __init__(self, base_clone_path: str, clone_subdirectory_prefix: str):
 
         self._base_clone_path = os.path.expanduser(base_clone_path)
         self._clone_subdirectory_prefix = clone_subdirectory_prefix
+        self._use_fixed_path = use_fixed_path
 
-    def get_clone_subdirectories(self) -> Generator[str, None, None]:
+    def _get_random_clone_subdirectories(self) -> Generator[str, None, None]:
         """
-        a generator yielding all the subdirectories of the base clone path
+        a generator yielding all the randomized subdirectories of the base clone path
         that are matching the clone pattern.
 
         Yields:
             the next subdirectory matching the pattern
         """
-        folders_with_pattern = get_filepaths_with_glob(self._base_clone_path, f"{self._clone_subdirectory_prefix}*")
+        folders_with_pattern = get_filepaths_with_glob(self._base_clone_path, f"{self._clone_subdirectory_prefix}-*")
         for folder in folders_with_pattern:
             yield folder
 
-    def get_single_clone_path(self) -> Optional[str]:
+    def _get_single_existing_random_clone_path(self) -> Optional[str]:
         """
-        searches for the single clone subdirectory in existance.
+        searches for the single randomly-suffixed clone subdirectory in existance.
 
         If found no such subdirectory or if found more than one (multiple matching subdirectories) - will return None.
         otherwise: will return the single and only clone.
         """
-        subdirectories = list(self.get_clone_subdirectories())
+        subdirectories = list(self._get_random_clone_subdirectories())
         if len(subdirectories) != 1:
             return None
         return subdirectories[0]
 
-    def create_new_clone_path(self) -> str:
+    def _generate_randomized_clone_path(self) -> str:
         folder_name = f"{self._clone_subdirectory_prefix}-{uuid.uuid4().hex}"
         full_local_repo_path = os.path.join(self._base_clone_path, folder_name)
+        return full_local_repo_path
+
+    def _get_fixed_clone_path(self) -> str:
+        return os.path.join(self._base_clone_path, self._clone_subdirectory_prefix)
+    
+    def get_clone_path(self) -> Optional[str]:
+        """
+        Get the clone path (fixed or randomized) if it exists
+        """
+        if self._use_fixed_path:
+            fixed_path = self._get_fixed_clone_path()
+            if os.path.exists(fixed_path):
+                return fixed_path
```

Comment:
why not simply `return self._get_fixed_clone_path()?`

if the path does not exist you are defaulting to the randomized path. this can't be good.

Reply:
No - it would return `None` (gets to the randomized path only under the `else` clause).
But this is an indication that it's not readable enough (I debated between the short and the readable version, went with the format, probably a mistake).

gonna fix to:
```
            if os.path.exists(fixed_path):
                return fixed_path
            else:
                return None
```